Panic: "Index out of range" when resizing the window very small.

**Describe the bug**
Panic when the window is resized to very small dimensions.

**To Reproduce**
Steps to reproduce the behavior:
1. `go get github.com/liamg/aminal`
2. Run `aminal`
3. Resize window as small as possible

**Expected behavior**
Avoid panic, eg by setting a minimum window size.

**Screenshots**
![screen shot 2018-11-28 at 8 27 47 pm](https://user-images.githubusercontent.com/6916093/49136175-bfc21880-f34d-11e8-8bd0-9e3224c41982.jpeg)
![screen shot 2018-11-28 at 8 28 04 pm](https://user-images.githubusercontent.com/6916093/49136178-c18bdc00-f34d-11e8-995c-af37f63896d7.jpeg)
![screen shot 2018-11-28 at 8 28 13 pm](https://user-images.githubusercontent.com/6916093/49136182-c3ee3600-f34d-11e8-8919-8655a1b23423.jpeg)


**Environment (please complete the following information):**
 - MacOS Mojave 10.14.2
 - go version go1.11.2 darwin/amd64
 - Zsh

**Logs**
```sh
panic: runtime error: index out of range

goroutine 35 [running]:
github.com/liamg/aminal/buffer.(*Buffer).Index(0xc00014e0b0)
   /Users/jack/go/src/github.com/liamg/aminal/buffer/buffer.go:567 +0x316
github.com/liamg/aminal/buffer.(*Buffer).NewLine(0xc00014e0b0)
   /Users/jack/go/src/github.com/liamg/aminal/buffer/buffer.go:745 +0x57
github.com/liamg/aminal/terminal.newLineSequenceHandler(0xc0014a4000, 0xc00014e2c0, 
0xa, 0xc000120038)
   /Users/jack/go/src/github.com/liamg/aminal/terminal/output.go:29 +0x40
github.com/liamg/aminal/terminal.(*Terminal).processInput(0xc00014e2c0, 0x48df2e0, 0
xc00009c140, 0xc0014a4000)
   /Users/jack/go/src/github.com/liamg/aminal/terminal/output.go:96 +0x35c
created by github.com/liamg/aminal/terminal.(*Terminal).Read
   /Users/jack/go/src/github.com/liamg/aminal/terminal/terminal.go:261 +0x1f9
```

--- a/gui/gui.go
+++ b/gui/gui.go
@@ -55,7 +55,7 @@ func New(config *config.Config, terminal *terminal.Terminal, logger *zap.Sugared
 
 // inspired by https://kylewbanks.com/blog/tutorial-opengl-with-golang-part-1-hello-opengl
 
-func (gui *GUI) scale() float32{
+func (gui *GUI) scale() float32 {
 	pw, _ := gui.window.GetFramebufferSize()
 	ww, _ := gui.window.GetSize()
 	return float32(ww) / float32(pw)
@@ -248,7 +248,7 @@ func (gui *GUI) Render() error {
 					}
 					if cell.Image() != nil {
 						gui.renderer.DrawCellImage(cell, uint(x), uint(y))
-					}else{
+					} else {
 						gui.renderer.DrawCellBg(cell, uint(x), uint(y), cursor, colour, false)
 					}
 				}
@@ -337,6 +337,7 @@ func (gui *GUI) createWindow(width int, height int) (*glfw.Window, error) {
 	if err != nil {
 		return nil, fmt.Errorf("Failed to create window: %s", err)
 	}
+	window.SetSizeLimits(300, 150, 10000, 10000)
 	window.MakeContextCurrent()
 
 	return window, nil

